addons/web/static/src/legacy/js/core/menu.js

```diff
@@ -59,11 +63,29 @@ export async function initAutoMoreMenu(el, options) {
         _adapt();
     }
 
-    const debouncedAdapt = () => {
-        clearTimeout(debounce);
-        debounce = setTimeout(_adapt, 250);
+    let pending = false;
+    let refreshId = null;
```

Comment:
I am not sure about your implementation still. Why do you need both these variables? Can't it work with `pending === !!refreshId` at all times?

Reply:
@qsm-odoo Actually, `pending !== false` means **"an `_adapt()` needs to be executed for the next browser refresh"** while `refreshId !== null` only means that **"an `onRefresh()` is programmed for the next browser refresh"**. Most of the time they mean the same thing, but sometimes they can be different. E.g:

For simplification, let's suppose the browser refresh is happening every **100ms** and `throttleAdapt()` is called on every "resize" event:

**Case [1]:** two "resize" events (at T(0) and T(50ms)):

```text
T(0ms)  : call `throttleAdapt()` => `refreshId !== null`, `pending === false`, call `_adapt()`.
T(50ms) : call `throttleAdapt()` => `refreshId !== null`, `pending !== false`.
T(100ms): call `onRefresh()`     => `refreshId !== null`, `pending === false`, call `_adapt()`. [A]
T(200ms): call `onRefresh()`     => `refreshId === null`
```

**Case [2]:** only one "resize" event (at T(0)):

```text
T(0ms)  : call `throttleAdapt()` => `refreshId !== null`, `pending === false`, call `_adapt()`.
T(100ms): call `onRefresh()`     => `refreshId === null` [B]
```

**[A]:** Here at the first refresh, we have a pending `_adapt()` to execute (from T(50ms)) that's why we have `pending === !!refreshId`.
**[B]:** In this case, the first and only `_adapt()` was executed immediately (`pending !== !!refreshId`), we just need to reset `refreshId` to `null`.

Reply:
Ok I see, I got confused into thinking there was a call to _adapt at every onRefresh call but that's not the case at all :shrug: This seems actually very logical now that I re-read it... maybe lacking some comments, I'll test and add comments myself if needed :+1: